src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/ScalarWaveKerrSchild.hpp

```diff
@@ -0,0 +1,225 @@
+  ScalarWaveKerrSchild& operator=(const ScalarWaveKerrSchild& /*rhs*/) = delete;
+  ScalarWaveKerrSchild(ScalarWaveKerrSchild&& /*rhs*/) noexcept = default;
+  ScalarWaveKerrSchild& operator=(ScalarWaveKerrSchild&& /*rhs*/) noexcept =
+      default;
+  ~ScalarWaveKerrSchild() = default;
+
+  // Tags specific to the solution
+  using scalar_wave_tags = tmpl::list<Pi, Phi<volume_dim>, Psi>;
+  using scalar_wave_dt_tags = db::wrap_tags_in<::Tags::dt, scalar_wave_tags>;
+
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataVector>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using spacetime_tags = tmpl::list<
+      gr::Tags::Lapse<DataVector>, ::Tags::dt<gr::Tags::Lapse<DataVector>>,
+      DerivLapse, gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>>,
+      DerivShift,
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      ::Tags::dt<
+          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>>,
+      DerivSpatialMetric, gr::Tags::SqrtDetSpatialMetric<DataVector>,
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataVector>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataVector>>;
+
+  using tags = tmpl::append<spacetime_tags, scalar_wave_tags>;
+
+  /// Returns curved scalar wave variables
+  tuples::tagged_tuple_from_typelist<scalar_wave_tags> variables(
+      const tnsr::I<DataVector, volume_dim>& x, scalar_wave_tags /*meta*/) const
+      noexcept;
+  /// \cond
+  /// Returns 3+1 spacetime variables and their time derivatives
+  tuples::tagged_tuple_from_typelist<spacetime_tags> variables(
+      const tnsr::I<DataVector, volume_dim>& x, spacetime_tags /*meta*/) const
+      noexcept;
+  /// \endcond
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+  SPECTRE_ALWAYS_INLINE double mass() const noexcept {
+    return kerr_schild_soln_.mass();
+  }
+  SPECTRE_ALWAYS_INLINE const std::array<double, volume_dim>& center() const
+      noexcept {
+    return kerr_schild_soln_.center();
+  }
+  SPECTRE_ALWAYS_INLINE const std::array<double, volume_dim>&
+  dimensionless_spin() const noexcept {
+    return kerr_schild_soln_.dimensionless_spin();
+  }
+
+ private:
+  template <typename LocalScalarFieldSolution>
+  friend bool operator==(
+      const ScalarWaveKerrSchild<LocalScalarFieldSolution>& lhs,
+      const ScalarWaveKerrSchild<LocalScalarFieldSolution>& rhs) noexcept;
+
+  template <typename LocalScalarFieldSolution>
+  friend bool operator!=(
```

Comment:
`!=` does not need to be a `friend` because it just calls `not(==)`

Reply:
`!=` changed to nonmember overload